src/analyzer/protocol/ldap/ldap.evt

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021 by the Zeek Project. See LICENSE for details.
+
+protocol analyzer LDAP_TCP over TCP:
+    parse with LDAP::Messages,
+    ports { 389/tcp, 3268/tcp};
```

Comment:
I haven't looked much around, but this one came to mind (and slightly related to the enable option).

I had removed `ports` usage in the QUIC `.evt` mainly due to re-compiling to add or remove a port seemed a weird workflow when it can be done in script land just as well. Wonder if you have thoughts on generally promoting the previous pattern rather than `ports` in .evt files. Is there an API to prevent one of the port in an .evt file to be used?
 
Compare with:
https://github.com/zeek/zeek/blob/0b940d8b3ce8ebb5539b99b8eea460fd6eac3cdc/scripts/base/protocols/http/main.zeek#L150-L161

Reply:
Doesn't `likely_server_ports` require reliable DPD signatures? With `ports` in the EVT file the analyzer would automatically be registered for a port, regardless of whether any of its signatures match (or if has any signatures at all).

Reply:
Yupp, that's the API the Spicy plugin [uses as well](https://github.com/zeek/zeek/blob/0b940d8b3ce8ebb5539b99b8eea460fd6eac3cdc/src/spicy/manager.cc#L694) so using `Analyzer::register_for_port(s)` should work, and be much easier redef'able if it was in script land (in particular since there seems to be no matching `unregister_for_port` for `Manager::UnregisterAnalyzerForPort`.

Reply:
I suggest to move the integrated (and possibly packaged version) of the analyzer to have ports/registration done in Zeek scripts then, even if it's more verbose. Besides redef, it also allows "in-a-pinch" hacking of base scripts to add/remove ports without needing to recompile.

May even consider adapting the package-template to include the necessary boiler plate and retire `ports` from `.evt` there.

@rsmmr - do you have differing opinions/thoughts?

Reply:
Yeah, agree, doing this in script-land is preferable. I added the EVT support early on just for convenience, but it really doesn't play along well with Zeek's normal analyzer handling.

Reply:
I have moved the port configuration to script land.